### PR TITLE
Replacing the ock-demo workflow to use multiple jobs for its components instead of using the same workflow.

### PR DESCRIPTION
**Pull request by MaryaSharf**
# Overview

Replacing the workflow of creating oneapi demo to use multiple jobs for its components instead of using the same workflow.

### CODE REVIEW

**Comment by MaryaSharf** [comment]
I'll remove the trigger on PR once it runs completely and successfully. 
Next step is to update the getting_started guide.

**Review by coldav — APPROVED** [review]

File: .github/actions/build_portBLAS_action/action.yml [hunk 1]
```diff
@@ -17,12 +17,13 @@ runs:
     - name: Set up Environment and Build portBLAS
       shell: bash
       run: |
-        ls
-        export LD_LIBRARY_PATH=${{ inputs.workspace }}/linux_nightly_release/lib:$LD_LIBRARY_PATH
-        export CMAKE_CXX_COMPILER=${{ inputs.workspace }}/linux_nightly_release/bin/clang++
-        export CMAKE_C_COMPILER=${{ inputs.workspace }}/linux_nightly_release/bin/clang
+        export LD_LIBRARY_PATH=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/lib/libsycl.so:${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/lib:$LD_LIBRARY_PATH
```
Comment (frasercrmck) [inline .github/actions/build_portBLAS_action/action.yml:20]: Does setting `LD_LIBRARY_PATH` to a specific `.so` actually do anything? I've have thought that pointing it to `/lib` as you're also doing would be enough.

File: .github/actions/build_portDNN_action/action.yml [hunk 2]
```diff
@@ -17,11 +17,12 @@ runs:
     - name: Set up Environment and build portDNN
       shell: bash
       run: |
-        ls
-        export LD_LIBRARY_PATH=${{ inputs.workspace }}/linux_nightly_release/lib:$LD_LIBRARY_PATH
-        export CMAKE_CXX_COMPILER=${{ inputs.workspace }}/linux_nightly_release/bin/clang++
-        export CMAKE_C_COMPILER=${{ inputs.workspace }}/linux_nightly_release/bin/clang
+        export LD_LIBRARY_PATH=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/lib/libsycl.so:${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/lib:$LD_LIBRARY_PATH
```
Comment (frasercrmck) [inline .github/actions/build_portDNN_action/action.yml:20]: Same question about the `.so`

File: .github/actions/build_portDNN_action/action.yml [hunk 3]
```diff
@@ -17,11 +17,12 @@ runs:
...
+        export CMAKE_CXX_COMPILER=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/bin/compiler/clang++
+        export CMAKE_C_COMPILER=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/bin/compiler/clang
         export OCL_ICD_FILENAMES=${{ inputs.workspace }}/install/lib/libCL.so
+        export DPCPP_INCLUDE_DIR=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/include/sycl
+        export PATH=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/include/sycl:${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/bin:${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/lib:$PATH
 
-        cmake -B portDNN_build_dir portDNN -GNinja -DCMAKE_CXX_COMPILER=$(pwd)/linux_nightly_release/bin/clang++ -DSNN_BUILD_BENCHMARKS=OFF -DSNN_BENCH_SYCLBLAS=OFF -DSNN_BUILD_DOCUMENTATION=OFF
+        cmake -B portDNN_build_dir portDNN -GNinja -DCMAKE_CXX_COMPILER=${{ inputs.workspace }}/intel/oneapi/compiler/2024.0/bin/compiler/clang++ -DSNN_BUILD_BENCHMARKS=OFF -DSNN_BENCH_SYCLBLAS=OFF -DSNN_BUILD_DOCUMENTATION=OFF
```
Comment (frasercrmck) [inline .github/actions/build_portDNN_action/action.yml:27]: You're exporting `CMAKE_CXX_COMPILER` above to the same value as you're setting `-DCMAKE_CXX_COMPILER` here. Could you just have, say, `-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}`? That's not getting into the question of whether both are actually required to be set or not.

File: .github/actions/build_portBLAS_action/action.yml (hunk 1 above)
Comment (coldav) [reply .github/actions/build_portBLAS_action/action.yml]: I think you are right, it's just a path.

**Pull request merged**